### ISSUE TITLE
Single crafting action produces two tree taps

### ISSUE THREAD

**Issue by mpatsift** [StephenGss/PAL#77]
In this nonov tournament:

**Comment by StephenGss** [comment]
I know this is old now, but could you send me the entire log for that. I'm wondering if the agent had another tree tap before this and the bug is that it isn't showing up properly in the inventory. The first line shows a tree_tap in the inventory with a count of zero.

**Comment by mpatsift** [comment]
Here you are
[2022-11-21-170029-openmind0-pal-tournament.log.gz](https://github.com/StephenGss/PAL/files/10385815/2022-11-21-170029-openmind0-pal-tournament.log.gz)

**Comment by StephenGss** [comment]
I've looked through the log and It's clear why there are two tree taps. 
At 17:15:35 the agent crafts a tree tap, then tries to place it. For some reason the placement failed and I think this is where PAL messed up. I believe it didn't update the inventory correctly so the agent couldn't see the tree tap still in the inventory.

Then the agent crafts another tree tap at 17:15:42 and after crafting the inventory gets updated correctly. 

I am still trying to figure out why the placement didn't work.

**Comment by mpatsift** [comment]
Yes, you're right.  I see the earlier craft and failed place.  And after the place fails tree tap agent inventory is zero and there's no tree tap in the environment.  However, it looks like polycraft is still telling our agent that the tree tap is the selected item.  Then when the agent crafts a new tree tap inventory jumps straight from zero to two.

**Comment by StephenGss** [comment]
I've run through the same config with the same actions and it seems this happens when the agent tries to place a tree tap right where the pogoist is teleporting to.  It's possible this was fixed between when it first occurred and the 36M eval. This is also an old config that shouldn't be used anymore as one of the traders is inside the wall.
